pkg/api/validation/validation.go

```diff
@@ -4018,14 +4018,14 @@ func ValidateResourceRequirements(requirements *api.ResourceRequirements, fldPat
 		// Check that request <= limit.
 		limitQuantity, exists := requirements.Limits[resourceName]
 		if exists {
-			// For GPUs, not only requests can't exceed limits, they also can't be lower, i.e. must be equal.
+			// For GPUs and huge pages, not only requests can't exceed limits, they also can't be lower, i.e. must be equal.
 			if quantity.Cmp(limitQuantity) != 0 && !helper.IsOvercommitAllowed(resourceName) {
-				allErrs = append(allErrs, field.Invalid(reqPath, quantity.String(), fmt.Sprintf("must be equal to %s limit", api.ResourceNvidiaGPU)))
+				allErrs = append(allErrs, field.Invalid(reqPath, quantity.String(), fmt.Sprintf("must be equal to %s limit", resourceName)))
```

Comment:
> A pod could be created with requests of resource Huge page, but no limits of it.

If so, I don't think this change will work, since limit is still not set actually. You're just skipping the comparison.

> That breaks the restriction that requests of hugepages should be equals to that of limits.

Where is the doc? I don't find this. /cc @vishh 

The [only doc](https://kubernetes.io/docs/tasks/manage-hugepages/scheduling-hugepages/#api) I found is,

> Huge page requests must equal the limits. This is the default if limits are specified, but requests are not.

"Huge page requests must equal the limits." has conditions.

Actually this is true for all kinds of resources, if limits are specified, but requests are not, default requests to limits.

Reply:
"_Huge page requests must equal the limits._" is a restriction, and statement "_This is the default if limits are specified, but requests are not._" is default behavior of the system, mainly as an additional introduction to users, I couldn't see any relationship between the two.

>If so, I don't think this change will work, since limit is still not set actually. You're just skipping the comparison.

the check contains in func **IsOvercommitAllowed**, just a fix on error info here.